### PR TITLE
issue: 848172 Fix coverity #24: Assigned value is garbage [hash_map]

### PR DESCRIPTION
**Pull request by DanielLibenson**
#24 (CLANG_WARNING) util/hash_map.inl:145:12: warning: Assigned value is garbage

Signed-off-by: Daniel Libenson <daneilli@mellanox.com>

### CODE REVIEW

File: src/vma/dev/ring.h [hunk 1]
```diff
@@ -79,19 +79,25 @@ typedef enum {
 } cq_type_t;
 
 /* udp uc key, only by destination port as we already know the rest */
-typedef struct __attribute__((packed)) {
+typedef struct __attribute__((packed)) flow_spec_udp_uc_key_t{
 	in_port_t 	dst_port;
```
Comment (igor-ivanov) [inline src/vma/dev/ring.h]: please add use `flow_spec_udp_uc_key_t {`

File: src/vma/dev/ring.h [hunk 2]
```diff
@@ -79,19 +79,25 @@ typedef enum {
...
+	flow_spec_udp_uc_key_t(): dst_port(INPORT_ANY){}//Default constructor
```
Comment (igor-ivanov) [inline src/vma/dev/ring.h]: please use `dst_port(INPORT_ANY) {}`

File: src/vma/dev/ring.h [hunk 3]
```diff
@@ -79,19 +79,25 @@ typedef enum {
...
+	explicit flow_spec_udp_uc_key_t(in_port_t d_port): dst_port(d_port){}//Constructor
 } flow_spec_udp_uc_key_t;
 
-typedef struct __attribute__((packed)) {
+typedef struct __attribute__((packed)) flow_spec_udp_mc_key_t{
```
Comment (igor-ivanov) [inline src/vma/dev/ring.h]: please use `flow_spec_udp_mc_key_t {`

File: src/vma/dev/ring.h [hunk 4]
```diff
@@ -79,19 +79,25 @@ typedef enum {
...
 	in_addr_t	dst_ip;
 	in_port_t	dst_port;
+	flow_spec_udp_mc_key_t(): dst_ip(INADDR_ANY), dst_port(INPORT_ANY){}//Default constructor
+	flow_spec_udp_mc_key_t(in_addr_t d_ip, in_addr_t d_port): dst_ip(d_ip), dst_port(d_port){}//Constructor
 } flow_spec_udp_mc_key_t;
 
-typedef struct __attribute__((packed)) {
+typedef struct __attribute__((packed)) flow_spec_tcp_key_t{
```
Comment (igor-ivanov) [inline src/vma/dev/ring.h]: please use `flow_spec_tcp_key_t {`

File: src/vma/dev/ring.h [hunk 5]
```diff
@@ -79,19 +79,25 @@ typedef enum {
...
 	in_addr_t	src_ip;
 	in_port_t	dst_port;
 	in_port_t	src_port;
+	flow_spec_tcp_key_t(): src_ip(INADDR_ANY), dst_port(INPORT_ANY), src_port(INPORT_ANY){}//Default constructor
+	flow_spec_tcp_key_t(in_addr_t	s_ip, in_addr_t	d_port, in_addr_t s_port): src_ip(s_ip), dst_port(d_port), src_port(s_port){}//Constructor
```
Comment (igor-ivanov) [inline src/vma/dev/ring.h]: please use `src_port(s_port) {} //Constructor`

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/414/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/428/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/416/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/430/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/417/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/431/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Merged https://github.com/Mellanox/libvma/pull/190

**Pull request closed without merging**